tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/publishResultCli.ts

```diff
@@ -0,0 +1,222 @@
+        const taskResultsPathArray = JSON.parse(resultConfig.taskResultsPath);
+        const taskResults: TaskResult[] = [];
+
+        for (const taskResultPath of taskResultsPathArray) {
+            if (fs.existsSync(taskResultPath)) {
+                taskResults.push(requireJsonc(taskResultPath));
+            } else {
+                logger.error(`SendSdkGenerationToDB failed !, ${taskResultPath} isn't exist`);
+            }
+        }
+
+        await publisher.sendSdkTaskResultToDB(resultConfig.pipelineBuildId, taskResults);
+    }
+
+    await publisher.close();
+}
+
+function getTrigger(config: ResultPublisherEventHubInput): Trigger {
+    let trigger: Trigger;
+    try {
+        trigger = JSON.parse(config.trigger);
+    } catch (error) {
+        logger.error(`Wrong json format:` + config.trigger);
+        throw new Error(error);
+    }
+
+    return trigger;
+}
+
+function getTaskResults(taskResultsPath: string): TaskResult[] {
+    const taskResultsPathArray = JSON.parse(taskResultsPath);
+    const taskResults: TaskResult[] = [];
+    for (const taskResultPath of taskResultsPathArray) {
+        if (fs.existsSync(taskResultPath)) {
+            taskResults.push(requireJsonc(taskResultPath));
+        } else {
+            logger.warn(`${taskResultPath} isn't exist, skip.`);
+        }
+    }
+    return taskResults;
+}
+
+async function publishEventhub(pipelineStatus: SDKPipelineStatus) {
+    resultPublisherEventHubInput.validate();
+    const config: ResultPublisherEventHubInput = resultPublisherEventHubInput.getProperties();
+    const trigger: Trigger = getTrigger(config);
+    let event: PipelineRunEvent = undefined;
+    const publisher: ResultEventhubPublisher = new ResultEventhubPublisher(config.eventHubConnectionString);
+
+    switch (pipelineStatus) {
+        case 'queued':
+            event = {
+                status: 'queued',
+                trigger: trigger,
+                pipelineBuildId: config.pipelineBuildId,
+            } as QueuedEvent;
+            break;
+        case 'in_progress':
+            event = {
+                status: 'in_progress',
+                trigger: trigger,
+                pipelineBuildId: config.pipelineBuildId,
+            } as InProgressEvent;
+            break;
+        case 'completed':
+            if (!config.resultsPath || !config.logPath) {
+                throw new Error(`Invalid completed event parameter!`);
+            }
+
+            const taskResults: TaskResult[] = getTaskResults(config.resultsPath);
+            const taskTotalResult: TaskResult = publisher.generateTotalResult(taskResults, config.pipelineBuildId);
+            event = {
+                status: 'completed',
+                trigger: trigger,
+                pipelineBuildId: config.pipelineBuildId,
+                logPath: config.logPath,
+                result: taskTotalResult,
+            } as CompletedEvent;
+            break;
```

Comment:
should add 'default' block to handle the wrong status?

Reply:
done